google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -288,6 +288,35 @@ public boolean next() throws SpannerException {
           }
         }
 
+        @Override
+        public boolean initiateStreaming(
+            AsyncResultSet.StreamMessageListener streamMessageListener) {
+          try {
+            boolean ret = super.initiateStreaming(streamMessageListener);
+            if (beforeFirst) {
+              synchronized (lock) {
+                session.get().markUsed();
+                beforeFirst = false;
```

Comment:
I don't think that this is correct. `beforeFirst` indicates whether the current cursor of the result set is positioned before the first row. Starting the stream would not move the cursor to the first row.

Reply:
Also, `session.get().markUsed()` and `sessionUsedForQuery` are set to true here, also if `super.initiateStreaming()` returned false. Is that correct? If it returned false, then that is an indication that the result set was not a streaming result set, which again indicated that the call was a no-op, right?

Reply:
Fixed,